install-server.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+set -e
+
+readonly ARGS=( "$@" )
+readonly PROGNAME="./install-server.sh"
+
+usage() {
+    cat <<- EOF
+    usage: $PROGNAME options
+
+    Install Trento Server
+
+    OPTIONS:
+       -p --private-key         the private key used by the runner to connect to the hosts
+       -h --help                show this help
+
+
+    Example:
+       $PROGNAME --private-key ./id_rsa_runner
+EOF
+}
+
+cmdline() {
+    local arg=
+    for arg
+    do
+        local delim=""
+        case "$arg" in
+            --private-key)  args="${args}-p ";;
+            --help)         args="${args}-h ";;
+            
+            *) [[ "${arg:0:1}" == "-" ]] || delim="\""
+            args="${args}${delim}${arg}${delim} ";;
+        esac
+    done
+    
+    eval set -- "$args"
+    
+    while getopts "ph" OPTION
+    do
+        case $OPTION in
+            h)
+                usage
+                exit 0
+            ;;
+            p)
+                readonly PRIVATE_KEY=$OPTARG
+            ;;
+            *)
+                usage
+                exit 0
+            ;;
+        esac
+    done
+
+    if [[ -z "$PRIVATE_KEY" ]]; then
+        read -rp "Please provide the path of the runner private key: " PRIVATE_KEY </dev/tty
```

Comment:
Maybe we should add a default option (~/.ssh/id_rsa)

Reply:
I'd advise against such a default otherwise a user might be annoyed that we are copying around his private key without confirmation

Reply:
I mean a default option that still requires the users confirmation 
`Please provide the path of the runner private key [~/.ssh/id_rsa]:`

Reply:
Putting myself from the user's perspective, I would definitely not like that.

What maybe would make sense for a future improvement would be to automatically create a new SSH key in case the user chose not to give us one, wdyt?

Reply:
Actually, I don't dislike Aleksei's suggestion. It is pretty common to have default values in the user intervention actions, to allow a quick click and go. 
@rtorrero I think you miss understood Aleksei's suggestion. He doesn't ask for a default value to not ask to the user, he just says that we might have a default value for the cases where you are asked, but you don't type anything, and simply click enter.

Reply:
hmm, this is a tricky discussion but also a gigantic bike shed. Let's follow-up later.